src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -90,9 +90,14 @@ private static boolean isValidPath(Path filePath) {
      * Saves all data to this storage file.
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * 
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
-
+    public void save(AddressBook addressBook) throws StorageOperationException{
+        
+        if(!path.toFile().isFile()){
+            throw new StorageOperationException("File not found: " + path);
```

Comment:
where will the thrown exception be handled? and how will your program handle it?